// @aquanow/aqua-config/configs/coins.js
/**
 * A list of coins that are being used across Aquanow Front-End and Back-End.
 *
 * IMPORTANT: The order of the coins must be in a specific order
 * in which it relies on the Front-End's User Balance Order
 *
 * @typedef {Object} coins
 * @property {string} coins.USDT - ISO 4217 currencies
 * @property {number} balancePrecision - To render decimal precision (balance) for each coin
 * @property {number} minimumSize - Minimum Trade Size Allowed
 * @property {number} pricePrecision - For rounding down withdraw/deposit limit and to render decimal precision on UI
 * @property {number} minimumSizeRFQ - For RFQ order algo processing
 * @property {number} [minimumWithdrawSize] - OPTIONAL for fiat, MANDATORY for crpyto to cover withdraw cost
 */
const coins = {
  USDT: {
    balancePrecision: 10,
    minimumSize: 10,
    pricePrecision: 8,
    minimumWithdrawSize: 1000,
    minimumSizeRFQ: 0.01
  },
  USD: {
    balancePrecision: 10,
    minimumSize: 10,
    pricePrecision: 2,
    minimumSizeRFQ: 0.01
  },
  CAD: {
    balancePrecision: 10,
    minimumSize: 10,
    pricePrecision: 2,
    minimumWithdrawSize: 10,
    minimumSizeRFQ: 0.01
  },
  EUR: {
    balancePrecision: 10,
    minimumSize: 10,
    pricePrecision: 2,
    minimumSizeRFQ: 0.01
  },
  GBP: {
    balancePrecision: 10,
    minimumSize: 10,
    pricePrecision: 2,
    minimumSizeRFQ: 0.01
  },
  AUD: {
    balancePrecision: 10,
    minimumSize: 10,
    pricePrecision: 2,
    minimumSizeRFQ: 0.01
  },
  BTC: {
    balancePrecision: 10,
    pricePrecision: 8,
    minimumSize: 0.002,
    minimumWithdrawSize: 0.001,
    minimumSizeRFQ: 0.0000001
  },
  ETH: {
    balancePrecision: 10,
    minimumSize: 0.15,
    pricePrecision: 8,
    minimumWithdrawSize: 0.02,
    minimumSizeRFQ: 0.0000001
  },
  XRP: {
    balancePrecision: 10,
    minimumSize: 100,
    pricePrecision: 2,
    minimumWithdrawSize: 25,
    minimumSizeRFQ: 1
  },
  LTC: {
    balancePrecision: 10,
    pricePrecision: 6,
    minimumSize: 0.5,
    minimumWithdrawSize: 0.01,
    minimumSizeRFQ: 0.0000001
  },
  EOS: {
    balancePrecision: 10,
    minimumSize: 10,
    minimumWithdrawSize: 1,
    pricePrecision: 4,
    minimumSizeRFQ: 0.0000001
  },
  BCH: {
    balancePrecision: 10,
    pricePrecision: 8,
    minimumSize: 0.15,
    minimumWithdrawSize: 0.002,
    minimumSizeRFQ: 0.0000001
  },
  XLM: {
    balancePrecision: 10,
    minimumSize: 500,
    minimumWithdrawSize: 3,
    pricePrecision: 8,
    minimumSizeRFQ: 1
  },
  LINK: {
    balancePrecision: 10,
    minimumSize: 1,
    pricePrecision: 4,
    minimumWithdrawSize: 1.4,
    minimumSizeRFQ: 0.0000001
  },
  BSV: {
    balancePrecision: 10,
    pricePrecision: 8,
    minimumSize: 0.15,
    minimumWithdrawSize: 0.0001,
    minimumSizeRFQ: 0.0000001
  }
};

module.exports = coins;
