// @aquanow/aqua-config/configs/exchangeList.js
const ibank = {
  tickers: {
    "EUR-USD": "EURUSD",
    "CAD-USD": "CADUSD",
    "CAD-EUR": "CADEUR"
  },
  walletIdSandbox: {
    EUR: "NTEzMzk",
    USD: "NTEzNDE",
    CAD: "NzQ1MDU"
  },
  walletID: {},
  taker: 0,
  cashout: 0
};

const binance = {
  tickers: {
    "BCH-USD": {
      market: "BCHPAX",
      exchange: "BCH/PAX"
    },
    "BTC-USD": {
      market: "BTCPAX",
      exchange: "BTC/PAX"
    },
    "EOS-USD": {
      market: "EOSPAX",
      exchange: "EOS/PAX"
    },
    "ETH-USD": {
      market: "ETHPAX",
      exchange: "ETH/PAX"
    },
    "LTC-USD": {
      market: "LTCPAX",
      exchange: "LTC/PAX"
    },
    "XLM-USD": {
      market: "XLMPAX",
      exchange: "XLM/PAX"
    },
    "XRP-USD": {
      market: "XRPPAX",
      exchange: "XRP/PAX"
    },
    "BCH-BTC": {
      market: "BCHBTC",
      exchange: "BCH/BTC"
    },
    "EOS-BTC": {
      market: "EOSBTC",
      exchange: "EOS/BTC"
    },
    "ETH-BTC": {
      market: "ETHBTC",
      exchange: "ETH/BTC"
    },
    "LTC-BTC": {
      market: "LTCBTC",
      exchange: "LTC/BTC"
    },
    "XLM-BTC": {
      market: "XLMBTC",
      exchange: "XLM/BTC"
    },
    "XRP-BTC": {
      market: "XRPBTC",
      exchange: "XRP/BTC"
    }
  },
  taker: 0.0,
  cashout: 0.0
};

const aquax = {
  tickers: {
    "BTC-CAD": {
      market: "BTC-CAD",
      exchange: "BTC-CAD"
    },
    "BTC-EUR": {
      market: "BTC-EUR",
      exchange: "BTC-EUR"
    },
    "BTC-GBP": {
      market: "BTC-GBP",
      exchange: "BTC-GBP"
    },
    "BTC-USD": {
      market: "BTC-USD",
      exchange: "BTC-USD"
    },
    "BTC-AUD": {
      market: "BTC-AUD",
      exchange: "BTC-AUD"
    },
  },
  taker: 0.0,
  cashout: 0.0
};

const poloniex = {
  tickers: {
    "BCH-USD": {
      market: 477,
      exchange: "BCH/USDC"
    },
    "BTC-USD": {
      market: 224,
      exchange: "BTC/USDC"
    },
    "EOS-USD": {
      market: 257,
      exchange: "EOS/USDC"
    },
    "ETH-USD": {
      market: 225,
      exchange: "ETH/USDC"
    },
    "LTC-USD": {
      market: 244,
      exchange: "LTC/USDC"
    },
    "XLM-USD": {
      market: 242,
      exchange: "XLM/USDC"
    },
    "XRP-USD": {
      market: 240,
      exchange: "XRP/USDC"
    },
    "BCH-BTC": {
      market: 475,
      exchange: "BCH/BTC"
    },
    "EOS-BTC": {
      market: 201,
      exchange: "EOS/BTC"
    },
    "ETH-BTC": {
      market: 148,
      exchange: "ETH/BTC"
    },
    "LTC-BTC": {
      market: 50,
      exchange: "LTC/BTC"
    },
    "XLM-BTC": {
      market: 89,
      exchange: "XLM/BTC"
    },
    "XRP-BTC": {
      market: 117,
      exchange: "XRP/BTC"
    }
  },
  taker: 0.0,
  cashout: 0.0
};

const kraken = {
  tickers: {
    "BTC-EUR": {
      market: "XBT/EUR",
      exchange: "BTC/EUR"
    }
  },
  taker: 0.26,
  cashout: 0.0
};

const b2c2 = {
  tickers: {
    "BTC-USDT": {
      market: "BTCUST.SPOT",
      exchange: "BTCUST.SPOT"
    },
    "BTC-USD": {
      market: "BTCUSD.SPOT",
      exchange: "BTCUSD.SPOT"
    },
    "LTC-USD": {
      market: "LTCUSD.SPOT",
      exchange: "LTCUSD.SPOT"
    },
    "ETH-USD": {
      market: "ETHUSD.SPOT",
      exchange: "ETHUSD.SPOT"
    },
    "BCH-USD": {
      market: "BCHUSD.SPOT",
      exchange: "BCHUSD.SPOT"
    },
    "XRP-USD": {
      market: "XRPUSD.SPOT",
      exchange: "XRPUSD.SPOT"
    },
    "EOS-USD": {
      market: "EOSUSD.SPOT",
      exchange: "EOSUSD.SPOT"
    },
    "XLM-USD": {
      market: "XLMUSD.SPOT",
      exchange: "XLMUSD.SPOT"
    },
    "BTC-CAD": {
      market: "BTCCAD.SPOT",
      exchange: "BTCCAD.SPOT"
    },
    "LTC-CAD": {
      market: "LTCCAD.SPOT",
      exchange: "LTCCAD.SPOT"
    },
    "ETH-CAD": {
      market: "ETHCAD.SPOT",
      exchange: "ETHCAD.SPOT"
    },
    "BCH-CAD": {
      market: "BCHCAD.SPOT",
      exchange: "BCHCAD.SPOT"
    },
    "XRP-CAD": {
      market: "XRPCAD.SPOT",
      exchange: "XRPCAD.SPOT"
    },
    "EOS-CAD": {
      market: "EOSCAD.SPOT",
      exchange: "EOSCAD.SPOT"
    },
    "XLM-CAD": {
      market: "XLMCAD.SPOT",
      exchange: "XLMCAD.SPOT"
    },
    "BTC-EUR": {
      market: "BTCEUR.SPOT",
      exchange: "BTCEUR.SPOT"
    },
    "LTC-EUR": {
      market: "LTCEUR.SPOT",
      exchange: "LTCEUR.SPOT"
    },
    "ETH-EUR": {
      market: "ETHEUR.SPOT",
      exchange: "ETHEUR.SPOT"
    },
    "BCH-EUR": {
      market: "BCHEUR.SPOT",
      exchange: "BCHEUR.SPOT"
    },
    "XRP-EUR": {
      market: "XRPEUR.SPOT",
      exchange: "XRPEUR.SPOT"
    },
    "ETH-BTC": {
      market: "ETHBTC.SPOT",
      exchange: "ETHBTC.SPOT"
    },
    "XRP-BTC": {
      market: "XRPBTC.SPOT",
      exchange: "XRPBTC.SPOT"
    },
    "LTC-BTC": {
      market: "LTCBTC.SPOT",
      exchange: "LTCBTC.SPOT"
    },
    "BCH-BTC": {
      market: "BCHBTC.SPOT",
      exchange: "BCHBTC.SPOT"
    },
    "EOS-BTC": {
      market: "EOSBTC.SPOT",
      exchange: "EOSBTC.SPOT"
    },
    "USDT-USD": {
      market: "USTUSD.SPOT",
      exchange: "USTUSD.SPOT"
    },
    "USDT-CAD": {
      market: "USTCAD.SPOT",
      exchange: "USTCAD.SPOT"
    },
    "USDT-EUR": {
      market: "USTEUR.SPOT",
      exchange: "USTEUR.SPOT"
    },
    "USDT-GBP": {
      market: "USTGBP.SPOT",
      exchange: "USTGBP.SPOT"
    },
    "BTC-GBP": {
      market: "BTCGBP.SPOT",
      exchange: "BTCGBP.SPOT"
    },
    "BTC-AUD": {
      market: "BTCAUD.SPOT",
      exchange: "BTCAUD.SPOT"
    },
  },
  levels: {
    "BTC-USDT": [2, 10, 25, 50, 100],
    "BTC-USD": [2, 10, 25, 50, 100],
    "LTC-USD": [50, 100, 150, 200, 250],
    "ETH-USD": [50, 100, 150, 200, 250],
    "BCH-USD": [25, 50, 75, 100, 125],
    "XRP-USD": [20000, 40000, 60000, 80000, 100000],
    "EOS-USD": [1000, 2000, 3000, 4000, 5000],
    "XLM-USD": [250000, 500000, 1000000, 2000000, 4000000],
    "BTC-CAD": [2, 10, 25, 50, 100],
    "LTC-CAD": [50, 100, 150, 200, 250],
    "ETH-CAD": [50, 100, 150, 200, 250],
    "BCH-CAD": [25, 50, 75, 100, 125],
    "XRP-CAD": [20000, 40000, 60000, 80000, 100000],
    "EOS-CAD": [1000, 2000, 3000, 4000, 5000],
    "XLM-CAD": [250000, 500000, 1000000, 2000000, 4000000],
    "BTC-EUR": [2, 10, 25, 50, 100],
    "LTC-EUR": [50, 100, 150, 200, 250],
    "ETH-EUR": [50, 100, 150, 200, 250],
    "BCH-EUR": [25, 50, 75, 100, 125],
    "XRP-EUR": [20000, 40000, 60000, 80000, 100000],
    "BTC-GBP": [2, 10, 25, 50, 100],
    "BTC-AUD": [2, 10, 25, 50, 100],
    "ETH-BTC": [50, 100, 150, 200, 250],
    "XRP-BTC": [20000, 40000, 60000, 80000, 100000],
    "LTC-BTC": [50, 100, 150, 200, 250],
    "BCH-BTC": [25, 50, 75, 100, 125],
    "EOS-BTC": [25, 50, 75, 100, 125],
    "USDT-USD": [10000, 100000, 200000, 300000, 500000],
    "USDT-CAD": [10000, 100000, 200000, 300000, 500000],
    "USDT-EUR": [10000, 100000, 200000, 300000, 500000],
    "USDT-GBP": [10000, 100000, 200000, 300000, 500000]
  },
  taker: 0.0,
  cashout: 0.0
};

const bitfinex = {
  tickers: {
    "BTC-USD": {
      market: "BTCUSD",
      exchange: "BTC/USD"
    },
    "LTC-USD": {
      market: "LTCUSD",
      exchange: "LTC/USD"
    },
    "ETH-USD": {
      market: "ETHUSD",
      exchange: "ETH/USD"
    },
    "BCH-USD": {
      market: "BCHUSD",
      exchange: "BCH/USD"
    },
    "XRP-USD": {
      market: "XRPUSD",
      exchange: "XRP/USD"
    },
    "XLM-USD": {
      market: "XLMUSD",
      exchange: "XLM/USD"
    },
    "EOS-USD": {
      market: "EOSUSD",
      exchange: "EOS/USD"
    },
    "ETH-BTC": {
      market: "ETHBTC",
      exchange: "ETH/BTC"
    },
    "EOS-BTC": {
      market: "EOSBTC",
      exchange: "EOS/BTC"
    },
    "XRP-BTC": {
      market: "XRPBTC",
      exchange: "XRP/BTC"
    },
    "LTC-BTC": {
      market: "LTCBTC",
      exchange: "LTC/BTC"
    },
    "BCH-BTC": {
      market: "BCHBTC",
      exchange: "BCH/BTC"
    },
    "XLM-BTC": {
      market: "XLMBTC",
      exchange: "XLM/BTC"
    }
  },
  taker: 0.2,
  cashout: 0.0
};

const bitstamp = {
  tickers: {
    "BTC-USD": "BTC/USD",
    "LTC-USD": "LTC/USD",
    "ETH-USD": "ETH/USD",
    "BCH-USD": "BCH/USD",
    "XRP-USD": "XRP/USD",
    "XLM-USD": "",
    "DASH-USD": "",
    "DOGE-USD": "",
    "EOS-USD": "",
    "ETH-BTC": "ETH/BTC",
    "XRP-BTC": "XRP/BTC",
    "LTC-BTC": "LTC/BTC",
    "BCH-BTC": "BCH/BTC",
    "XLM-BTC": "",
    "DASH-BTC": "",
    "DOGE-BTC": ""
  },
  taker: 0.25,
  cashout: 0.0
};

const coinbase = {
  tickers: {
    "BTC-USD": "BTC/USD",
    "LTC-USD": "LTC/USD",
    "ETH-USD": "ETH/USD",
    "BCH-USD": "BCH/USD",
    "XRP-USD": "",
    "XLM-USD": "",
    "DASH-USD": "",
    "DOGE-USD": "",
    "EOS-USD": "",
    "ETH-BTC": "ETH/BTC",
    "XRP-BTC": "",
    "LTC-BTC": "LTC/BTC",
    "BCH-BTC": "BCH/BTC",
    "XLM-BTC": "",
    "DASH-BTC": "",
    "DOGE-BTC": ""
  },
  taker: 0.25,
  cashout: 0.0
};

const gemini = {
  tickers: {
    "BTC-USD": "BTC/USD",
    "LTC-USD": "LTC/USD",
    "ETH-USD": "ETH/USD",
    "BCH-USD": "",
    "XRP-USD": "",
    "XLM-USD": "",
    "DASH-USD": "",
    "DOGE-USD": "",
    "EOS-USD": "",
    "ETH-BTC": "ETH/BTC",
    "XRP-BTC": "",
    "LTC-BTC": "LTC/BTC",
    "BCH-BTC": "",
    "XLM-BTC": "",
    "DASH-BTC": "",
    "DOGE-BTC": ""
  },
  taker: 1,
  cashout: 0.0
};

const huobipro = {
  tickers: {
    "BTC-USD": { market: "BTCUSDT", exchange: "BTC/USDT" },
    "ETH-USD": { market: "ETHUSDT", exchange: "ETH/USDT" },
    "LTC-USD": { market: "LTCUSDT", exchange: "LTC/USDT" },
    "XRP-USD": { market: "XRPUSDT", exchange: "XRP/USDT" },
    "DASH-USD": { market: "DASHUSDT", exchange: "DASH/USDT" },
    "ETH-BTC": { market: "ETHBTC", exchange: "ETH/BTC" },
    "LTC-BTC": { market: "LTCBTC", exchange: "LTC/BTC" },
    "XRP-BTC": { market: "XRPBTC", exchange: "XRP/BTC" },
    "DASH-BTC": { market: "DASHBTC", exchange: "DASH/BTC" }
  },
  taker: 0.0,
  cashout: 0.0
};

const bitmex = {
  tickers: {
    "BTC-USD": "BTC/USD",
    "LTC-USD": "",
    "ETH-USD": "ETH/USD",
    "BCH-USD": "",
    "XRP-USD": "",
    "XLM-USD": "",
    "EOS-USD": "",
    "ETH-BTC": "",
    "XRP-BTC": "",
    "LTC-BTC": "",
    "BCH-BTC": "",
    "XLM-BTC": ""
  },
  taker: 0.0,
  cashout: 0.0
};

const ftx = {
  tickers: {
    "BTC-USD": { market: "BTC/USD", exchange: "BTC/USD" },
    "ETH-USD": { market: "ETH/USD", exchange: "ETH/USD" },
    "LTC-USD": { market: "LTC/USD", exchange: "LTC/USD" },
    "BCH-USD": { market: "BCH/USD", exchange: "BCH/USD" },
    "LINK-USD": { market: "LINK/USD", exchange: "LINK/USD" }
  },
  taker: 0.0,
  cashout: 0.0
};

const circle = {
  tickers: {
    "BTC-USD": {
      market: "",
      exchange: "BTCUSD"
    },
    "LTC-USD": "",
    "ETH-USD": "",
    "BCH-USD": "",
    "XRP-USD": "",
    "XLM-USD": "",
    "DASH-USD": "",
    "DOGE-USD": "",
    "EOS-USD": "",
    "ETH-BTC": "",
    "XRP-BTC": "",
    "LTC-BTC": "",
    "BCH-BTC": "",
    "DASH-BTC": "",
    "DOGE-BTC": ""
  },
  taker: 0.0,
  cashout: 0.0
};

const falconx = {
  tickers: {
    "BTC-USD": {
      market: "",
      exchange: "BTC-USD"
    },
    "LTC-USD": {
      market: "",
      exchange: "LTC-USD"
    },
    "ETH-USD": {
      market: "",
      exchange: "ETH-USD"
    },
    "BCH-USD": {
      market: "",
      exchange: "BCH-USD"
    },
    "XRP-USD": {
      market: "",
      exchange: "XRP-USD"
    },
    "XLM-USD": {
      market: "",
      exchange: "XLM-USD"
    },
    "EOS-USD": {
      market: "",
      exchange: "EOS-USD"
    }
  },
  taker: 0.0,
  cashout: 0.0
};

const dvchain = {
  tickers: {
    "BTC-CAD": {
      market: "",
      exchange: "BTC-CAD"
    },
    "BCH-CAD": {
      market: "",
      exchange: "BCH-CAD"
    },
    "ETH-CAD": {
      market: "",
      exchange: "ETH-CAD"
    },
    "LTC-CAD": {
      market: "",
      exchange: "LTC-CAD"
    },
    "BTC-USD": {
      market: "",
      exchange: "BTC-USD"
    },
    "BCH-USD": {
      market: "",
      exchange: "BCH-USD"
    },
    "ETH-USD": {
      market: "",
      exchange: "ETH-USD"
    },
    "LTC-USD": {
      market: "",
      exchange: "LTC-USD"
    },
    "ETH-BTC": {
      market: "",
      exchange: "ETH-BTC"
    }
  },
  taker: 0.0,
  cashout: 0.0
};

const okex = {
  tickers: {
    "BTC-USD": {
      market: "BTC-USDT",
      exchange: "BTC/USDT"
    },
    "LTC-USD": {
      market: "LTC-USDT",
      exchange: "LTC/USDT"
    },
    "ETH-USD": {
      market: "ETH-USDT",
      exchange: "ETH/USDT"
    },
    "BCH-USD": {
      market: "BCH-USDT",
      exchange: "BCH/USDT"
    },
    "XRP-USD": {
      market: "XRP-USDT",
      exchange: "XRP/USDT"
    },
    "DASH-USD": {
      market: "DASH-USDT",
      exchange: "DASH/USDT"
    },
    "XLM-USD": {
      market: "XLM-USDT",
      exchange: "XLM/USDT"
    },
    "EOS-USD": {
      market: "EOS-USDT",
      exchange: "EOS/USDT"
    },
    "LTC-BTC": {
      market: "LTC-BTC",
      exchange: "LTC/BTC"
    },
    "ETH-BTC": {
      market: "ETH-BTC",
      exchange: "ETH/BTC"
    },
    "BCH-BTC": {
      market: "BCH-BTC",
      exchange: "BCH/BTC"
    },
    "XRP-BTC": {
      market: "XRP-BTC",
      exchange: "XRP/BTC"
    },
    "DASH-BTC": {
      market: "DASH-BTC",
      exchange: "DASH/BTC"
    },
    "XLM-BTC": {
      market: "XLM-BTC",
      exchange: "XLM/BTC"
    },
    "EOS-BTC": {
      market: "EOS-BTC",
      exchange: "EOS/BTC"
    }
  },
  taker: 0.0,
  cashout: 0.0
};

const cryptodotcom = {
  tickers: {
    "BTC-USD": {
      market: "BTC_USDT",
      exchange: "BTC_USDT"
    },
    "ETH-USD": {
      market: "ETH_USDT",
      exchange: "ETH_USDT"
    },
    "BCH-USD": {
      market: "BCH_USDT",
      exchange: "BCH_USDT"
    },
    "XRP-USD": {
      market: "XRP_USDT",
      exchange: "XRP_USDT"
    },
    "LTC-USD": {
      market: "LTC_USDT",
      exchange: "LTC_USDT"
    },
    "EOS-USD": {
      market: "EOS_USDT",
      exchange: "EOS_USDT"
    },
    "ETH-BTC": {
      market: "ETH_BTC",
      exchange: "ETH_BTC"
    },
    "XRP-BTC": {
      market: "XRP_BTC",
      exchange: "XRP_BTC"
    },
    "BCH-BTC": {
      market: "BCH_BTC",
      exchange: "BCH_BTC"
    }
  },
  taker: 0.0,
  cashout: 0.0
};

module.exports = {
  binance,
  kraken,
  poloniex,
  b2c2,
  bitfinex,
  bitstamp,
  huobipro,
  bitmex,
  ibank,
  gemini,
  ftx,
  circle,
  aquax,
  coinbase,
  falconx,
  dvchain,
  okex,
  cryptodotcom
};
