// @aquanow/aqua-config/configs/tickers.js
/**
 * A list of tickers/symbols that are being used across Aquanow Front-End and Back-End.
 
 * @typedef {Object} tickers
 * @property {string} tickers."BTC-USDT" - cryptocurrency tickers/symbols
 * @property {number} pricePrecision - To render decimal price precision for each ticker
 * @property {number} sizePrecision - To render size precision for each ticker
 * @property {number} marketImpactDepth - To calculate trade analysis for UI, depth used for market impact
 * @property {number} marketImpactSizePerDepth - To calculate trade analysis, size for each depth used for market impact
 * @property {number} maxMarketImpactDepth - To calculate trade analysis, max impact on market when buying/selling certain quantity
 * @property {boolean} hideFrontEnd - To hide ticker on front end
 */
const tickers = {
  "BTC-USDT": {
    pricePrecision: 2,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    hideFrontEnd: false
  },
  "BTC-USD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "ETH-USD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "XRP-USD": {
    pricePrecision: 5,
    sizePrecision: 1,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 10000,
    maxMarketImpactDepth: 500
  },
  "LTC-USD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "BCH-USD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "XLM-USD": {
    pricePrecision: 5,
    sizePrecision: 1,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "EOS-USD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "USD-CAD": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    fxTrade: true,
    hideFrontEnd: true
  },
  "USDT-USD": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "USD-EUR": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    fxTrade: true,
    hideFrontEnd: true
  },
  "USD-GBP": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    fxTrade: true,
    hideFrontEnd: true
  },
  "USD-BRL": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    fxTrade: true,
    hideFrontEnd: true
  },
  "USD-JPY": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    fxTrade: true,
    hideFrontEnd: true
  },
  "USD-MXN": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    fxTrade: true,
    hideFrontEnd: true
  },
  "USD-AUD": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    fxTrade: true,
    hideFrontEnd: true
  },
  "BTC-CAD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "ETH-CAD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "XRP-CAD": {
    pricePrecision: 5,
    sizePrecision: 1,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 10000,
    maxMarketImpactDepth: 500
  },
  "LTC-CAD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "BCH-CAD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "XLM-CAD": {
    pricePrecision: 5,
    sizePrecision: 1,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 10000,
    maxMarketImpactDepth: 500
  },
  "EOS-CAD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },

  "USDT-CAD": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    hideFrontEnd: false
  },

  "BTC-EUR": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },

  "ETH-EUR": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "XRP-EUR": {
    pricePrecision: 5,
    sizePrecision: 1,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 10000,
    maxMarketImpactDepth: 500
  },
  "LTC-EUR": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "BCH-EUR": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "USDT-EUR": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    hideFrontEnd: false
  },
  "ETH-BTC": {
    pricePrecision: 6,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "XRP-BTC": {
    pricePrecision: 8,
    sizePrecision: 0,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 100000,
    maxMarketImpactDepth: 500
  },
  "LTC-BTC": {
    pricePrecision: 6,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "BCH-BTC": {
    pricePrecision: 6,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "XLM-BTC": {
    pricePrecision: 8,
    sizePrecision: 0,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 100000,
    maxMarketImpactDepth: 500
  },
  "EOS-BTC": {
    pricePrecision: 6,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 1000,
    maxMarketImpactDepth: 500
  },
  "BTC-GBP": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "USDT-GBP": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    hideFrontEnd: false
  },
  "LINK-USD": {
    pricePrecision: 4,
    sizePrecision: 1,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    hideFrontEnd: false
  },
  "LINK-CAD": {
    pricePrecision: 4,
    sizePrecision: 1,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    hideFrontEnd: false
  },
  "BTC-AUD": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500
  },
  "BTC-INR": {
    pricePrecision: 2,
    sizePrecision: 4,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    hideFrontEnd: true
  },
  "USDT-INR": {
    pricePrecision: 4,
    sizePrecision: 2,
    marketImpactDepth: 20,
    marketImpactSizePerDepth: 50,
    maxMarketImpactDepth: 500,
    hideFrontEnd: true
  }
};

module.exports = tickers;
